File: 2018_kakao_blind_recruitment/비밀지도.js
function solution(n, arr1, arr2) {
  let answer = new Array(n).fill(new Array(""));
  let darr1 = new Array(n).fill(new Array(""));
  let darr2 = new Array(n).fill(new Array(""));

  for (let i = 0; i < n; i++) {
    darr1[i] = arr1[i].toString(2);
    darr2[i] = arr2[i].toString(2);
    if (darr1[i].length < n) {
      darr1[i] = "0".repeat(n - darr1[i].length) + darr1[i];
    }
    if (darr2[i].length < n) {
      darr2[i] = "0".repeat(n - darr2[i].length) + darr2[i];
    }
  }

  for (let i = 0; i < n; i++) {
    for (let j = 0; j < n; j++) {
      if (darr1[i][j] === "0" && darr2[i][j] === "0") {
        answer[i] += " ";
      } else {
        answer[i] += "#";
      }
    }
  }

  return answer;
}

File: 2018_kakao_blind_recruitment/비밀지도.java
import java.util.List;
import java.util.ArrayList;

// https://school.programmers.co.kr/learn/courses/30/lessons/17681

class Solution {
    
    public char[][] map;
    public String map1, map2;
    public List<String> answer;
    
    public String[] solution(int n, int[] arr1, int[] arr2) {
        answer = new ArrayList<String>();
        map = new char[n][n];
        
        for (int i = 0; i < n; i++) {
            map1 = translateBinary(n, arr1[i]);
            map2 = translateBinary(n, arr2[i]);
            for (int j = 0; j < n; j++) {
                if (map1.charAt(j) == '0' && map2.charAt(j) == '0') {
                    map[i][j] = ' ';
                } else {
                    map[i][j] = '#';
                }
            }
        }
        for (int i = 0; i < n; i++) {
            answer.add(new String(map[i]));
        }
        return answer.toArray(new String[answer.size()]);
    }
    public String translateBinary(int n, int target) {
        String str = Integer.toBinaryString(target);
        str = "0".repeat(n - str.length()) + str;
        return str;
    }
}
